### ISSUE TITLE
_rankN pepXML files

### ISSUE THREAD

**Issue by chambm** [Nesvilab/MSFragger#334]
Why does MSFragger in DIA mode write pepXML files with rank suffixes instead of using the hit_rank="N" attribute in a single pepXML? Is there an option to always write the DDA way?

**Comment by fcyu** [comment]
Two major reasons

1. AFAIK, each `spectrum_query` can only have one `precursor_neutral_mass` and `assumed_charge`, which is not suitable for DIA
2. Some downstream tools, such as PeptideProphet and Philosopher, don't read >1 ranked `search_hit`

Best,

Fengchao

**Comment by chambm** [comment]
True. Maybe multiple spectrum_query elements in that case? That is the way it's supposed to work for assumed_charge in the chimeric DDA case.

**Comment by fcyu** [comment]
Yes, multiple `spectrum_query` will work if the `spectrum`, `spectrumNativeID`, and `start_scan` can be non-unique. 

But then, how to specify the ranks of the same spectrum?

Best,

Fengchao

**Comment by chambm** [comment]
Yes, because of the assumed_charge thing they don't have to be unique. The ranks would be specific to a hypothetical precursor (theoretical mass and charge).

**Comment by fcyu** [comment]
Sorry that my previous question is not clear. How to specify the rank 1, 2, 3 for the `search_hit` of the same spectrum if list them in a separated `spectrum_query`? Can the `hit_rank` stars with > 1?

Thanks,

Fengchao

**Comment by chambm** [comment]
Each spectrum_query should start with hit_rank=1. Think of the rank as being for the hypothetical precursor ion rather than for the spectrum. Isn't that how it already works with the _rank1, _rank2 separate files?

**Comment by fcyu** [comment]
> Isn't that how it already works with the _rank1, _rank2 separate files?

In the _rank1, _rank2 files, we get the rank information by the file name. If we put all ranks in the same file and separate them in different `spectrum_query`, is there a way to mark different ranks? Or I have to rank them by the hyperscore when loading the data?

Thanks,

Fengchao

**Comment by chambm** [comment]
Yes, I suppose if you need to aggregate everything back at the spectrum level, you'll have to regenerate the ranks by whatever score you want to use. For percolator I'd expect to use its q-value to do the reranking so potentially something that was rank 2 will become rank 1.

**Comment by fcyu** [comment]
Thanks. Then, I need to make the changes and test if the downstream tools such as PeptideProphet and Philosopher support it.

Best,

Fengchao
